src/main/java/io/burt/jmespath/contrib/function/TranslateFunction.java

```diff
@@ -0,0 +1,58 @@
+package io.burt.jmespath.contrib.function;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import io.burt.jmespath.Adapter;
+import io.burt.jmespath.JmesPathType;
+import io.burt.jmespath.function.ArgumentConstraints;
+import io.burt.jmespath.function.FunctionArgument;
+
+public class TranslateFunction extends SubstringMatchingFunction {
+  public TranslateFunction() {
+    super(ArgumentConstraints.listOf(3, 3, ArgumentConstraints.typeOf(JmesPathType.STRING)));
+  }
+
+  @Override
+  protected <T> T callFunction(Adapter<T> runtime, List<FunctionArgument<T>> arguments) {
+    String arg = runtime.toString(arguments.get(0).value());
+    String map = runtime.toString(arguments.get(1).value());
+    String trans = runtime.toString(arguments.get(2).value());
+
+    if (isEmpty(arg)) {
+      return runtime.createString("");
+    } else {
+      return runtime.createString(replaceChars(arg, map, trans));
+    }
+  }
+
+  /**
+   * This is map-based reimplementation of org.apache.commons.lang.StringUtils.replaceChars
+   */
```

Comment:
If this is a copyright or license issue we need to think more about this, but if the code is sufficiently different I think you should just drop this comment. I don't see what it adds to know this information.

Reply:
Originally it was a note for me only when I realized that `jmespath-java` does not depend such libraries at all that our company usually does.

Though I am not a copyright expert at all IMHO this is so trivial piece of code and using map (which might be quite overkill) instead of looking up characters in the original string makes it different enough that it is by far not an infringement. 

Here is the link to the referenced implementation:
https://commons.apache.org/proper/commons-lang/apidocs/src-html/org/apache/commons/lang3/StringUtils.html#line.5992

So I removed the the comment.